File: content/en/04-llms/llms/_exercises/09_pdf/09_pdf.R
library(ellmer)

recipe_pdfs <- here::here("data/recipes/pdf")
pdf_waffles <- file.path(recipe_pdfs, "CinnamonPeachOatWaffles.pdf")

# Ask the model to turn this messy PDF print-out of a waffle
# recipe into a clean list of ingredients and steps to follow.
chat <- chat_posit(model = "claude-haiku-4-5")
chat$chat(
  "____",
  ____(pdf_waffles)
)

File: content/en/04-llms/llms/_exercises/09_pdf/09_pdf.py
# %%
import chatlas
from pyhere import here

# %%
recipe_pdfs = here("data/recipes/pdf/")
pdf_cheesesteak = recipe_pdfs / "PhillyCheesesteak.pdf"

# %% [markdown]
# Ask the model to turn this messy PDF print-out of a Philly Cheesesteak
# recipe into a clean list of ingredients and steps to follow.

# %%
chat = chatlas.ChatPosit(model="claude-haiku-4-5")
chat.chat(
    "____",
    chatlas.____(pdf_cheesesteak),
)
